### PR TITLE
KWS-2137: Update Intro view with new MVVM structure

### PR DESCRIPTION
**Pull request by Tbruni85**
## JIRA
https://jira.superawesome.tv/browse/KWS-2137

## DESCRIPTION
*InitialView* refactored to the new MVVM structure

## SCREENSHOTS
simulator screen shot - iphone 6s - 2018-04-05 at 12 47 24

simulator screen shot - iphone x - 2018-04-05 at 12 46 38

### CODE REVIEW

**Comment by Tbruni85** [comment]
@devgabrielcoman 
For this view I preferred to leave the *ViewModel* injected to make it easier to init the View.

File: PopJamUI/Classes/InitialView/Views/InitialView.swift [hunk 1]
```diff
@@ -134,23 +136,51 @@ public class InitialView: UIViewController, UITextViewDelegate, ViewModelDelegat
...
+        let webView = WebView.init(withTitle: title, url: url)
+        let navController = UINavigationController.init(rootViewController: webView)
+        self.present(navController, animated: true, completion: nil)
     }
     
-    public func canPresentViewController(vc: UIViewController) {}
+    private func needToPushView(buttonTag: Int) {
+        
+        var vc: UIViewController
+        if buttonTag == 1 {
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/InitialView/Views/InitialView.swift]: Looks good!

But I wouldn't use tags anymore to figure out what button's been tapped.
Either assign two different @IBActions to each button or assign one, but do this:

```
@objc private func buttonTapped(sender: UIButton) {
   if sender is getStartedButton {
     // do action 1
   } 
   else if sender is loginButton {
     // do action 2
   }
}
```
Using tags is messier and doesn't really use Swift's type system as it should.

File: PopJamUI/Classes/InitialView/Views/InitialView.swift [hunk 2]
```diff
@@ -134,23 +136,51 @@ public class InitialView: UIViewController, UITextViewDelegate, ViewModelDelegat
...
+            let createUserNameDataProvider = CreateUserNameDataProvider()
+            let createUserNameViewModel = CreateUserNameViewModel.init(withDataProvider: createUserNameDataProvider)
+            vc = CreateUserNameView.init(withViewModel: createUserNameViewModel)
+        } else {
+            let authService = PopJamSDKService.getService(value: AuthServiceProtocol.self, environment: EUStagingEnvironment.init())
+            let config = PopJamSDKService.getService(value: ConfigServiceProtocol.self, environment: EUStagingEnvironment.init())
+            
+            config?.getConfig(completionHandler: { configModel, error in
+                
+                let dataProvider = LoginViewDataProvider.init(authService: authService!, config: configModel as! SocialConfigModelProtocol)
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/InitialView/Views/InitialView.swift]: Would like to have a discussion w/ Myles, Usmaan, Gui, etc. 
Do we keep logic like this in the View itself?
Or should we delegate to the (as yet non-existent) router? 

Cause I'd just call a View Delegate here saying:

```
delegate?.didTapOnLoginButton ()
// or
delegate?.didTapOnCreateButton ()
```

or 

```
router.display (Views.login.)
// or
router.display (Views.create.username)
```

But it's alright for now!

File: PopJamUI/Classes/InitialView/Views/InitialView.swift (hunk 1 above)
Comment (Tbruni85) [reply PopJamUI/Classes/InitialView/Views/InitialView.swift]: 👍

**Pull request merged**